### PR TITLE
Add basic kdoc to builder functions which explains presence of _ suffix

### PR DESCRIPTION
**Pull request by dlurton**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pig/resources/org/partiql/pig/templates/kotlin.ftl [hunk 1]
```diff
@@ -102,6 +102,15 @@ class ${t.kotlinName}(
 [#--Emits builder functions that wraps the constructors of domain type. --]
 [#macro builder_constructor_fun t return_type]
 [#list t.builderFunctions as bf]
+/**
+ * Creates an instance of [${return_type}].
+[#if bf.kotlinName?ends_with("_")]
+ *
+ * Use this variant when metas must be passed to primitive child elements.
+ *
+ * (The "_" suffix is needed to work-around conflicts due to type erasure and ambiguities nullable arguments.)
```
Comment (therapon) [inline pig/resources/org/partiql/pig/templates/kotlin.ftl:111]: ambiguities with null arguments

File: pig/src/org/partiql/pig/generator/kotlin/KTypeDomain.kt [hunk 2]
```diff
@@ -79,8 +79,6 @@ data class KTuple(
     val properties: List<KProperty>,
     val arity: IntRange,
     val builderFunctions: List<KBuilderFunction>,
-    /** The base name of the builder function. */
-    val builderName: String,
```
Comment (therapon) [inline pig/src/org/partiql/pig/generator/kotlin/KTypeDomain.kt:83]: Why is this removed?

File: pig/src/org/partiql/pig/generator/kotlin/KTypeDomainConverter.kt [hunk 3]
```diff
@@ -64,7 +64,6 @@ internal class KTypeDomainConverter(private val typeDomain: TypeDomain) {
             superClass = superClass,
             constructorName = constructorName,
             properties = computeProperties(this),
-            builderName = this.tag.snakeToCamelCase() + "_",
```
Comment (therapon) [inline pig/src/org/partiql/pig/generator/kotlin/KTypeDomainConverter.kt:67]: Same question here?

Comment (dlurton) [reply pig/src/org/partiql/pig/generator/kotlin/KTypeDomainConverter.kt:67]: `builderName` was unused actually.  Added to a previous PR.
